src/khepri_fun.erl

```diff
@@ -337,7 +337,26 @@ to_standalone_fun2(Fun, State) ->
         fun_kept ->
             {Fun, State};
         undefined ->
-            to_standalone_fun3(Fun, State)
+            undefined
+    end.
+
+to_standalone_fun2_with_retry(Fun, State) ->
+    case to_standalone_fun2(Fun, State) of
+        undefined ->
+            Lock = {{khepri_fun, Fun}, self()},
+            try
+                global:set_lock(Lock, [node()]),
+                case to_standalone_fun2(Fun, State) of
+                    undefined ->
+                        to_standalone_fun3(Fun, State);
+                    Other ->
+                        Other
+                end
+            after
+                global:del_lock(Lock, [node()])
+            end;
+        Other ->
+            Other
```

Comment:
Could you please name the variable `StandaloneFunAndState` for instance? I feel like `Other` could be any random garbage returned by the function, but we always expect a success return value.